src/pci/src/configuration.rs

```diff
@@ -698,108 +649,95 @@ impl PciConfiguration {
     /// Adds a region specified by `config`.  Configures the specified BAR(s) to
     /// report this region and size to the guest kernel.  Enforces a few constraints
     /// (i.e, region size must be power of two, register not already used).
-    pub fn add_pci_bar(&mut self, config: &PciBarConfiguration) -> Result<()> {
-        let bar_idx = config.idx;
+    pub fn add_pci_bar(&mut self, bar_idx: usize, addr: u64, size: u64) -> Result<(), PciError> {
         let reg_idx = BAR0_REG + bar_idx;
 
         if bar_idx >= NUM_BAR_REGS {
-            return Err(Error::BarInvalid(bar_idx));
+            error!("pci: invalid BAR index: {bar_idx}");
```

Comment:
These error logs are quite useless because this error will propagate all the way to the vm builder code and then stop the firecracker. Actually maybe we can use asserts here? This is an internal configuration code, so we should not add 'invalid' bars anyway.

Reply:
These are in the path of `/actions InstanceStart`, so I think we shouldn't assert. We should return an Error to the user, right? In that case, the error message in the logs is meant to help us debug more easily what went wrong.

Reply:
Mh, I thought the discussions we had internally were that for internal logic errors, we _do_ want to panic. That's definitely how I've been writing my code for the last few months :thinking:

Reply:
sure, but these asserts are triggered by the guest MMIO exits not along an API path. In any case, if indeed my understanding was wrong, I'm happy to turn these in `assert`ions/`unwrap`s

Reply:
hmm.. you are right, we have asserts in "guest mmio" path, but here we have error in "internal config" path, so we do error handling totally upside down so to speak... Here, since it is only internal configuration we should assert, while in "guest mmio" path we should have error propagation or better a validity checks up front and then assert inside PCI code.

Reply:
Ok. I've turned those errors in assertions and got rid of the error type all together. Needed to re-write some of the unit tests as well. So please take a look at these too.